kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/frame/DecodedRequestFrame.java

```diff
@@ -48,4 +49,8 @@ private boolean isZeroAcksProduceRequest() {
         return apiKeyId() == PRODUCE.id && ((ProduceRequestData) body).acks() == 0;
     }
 
+    public DecodedResponseFrame<?> responseFrame(ResponseHeaderData header, ApiMessage message) {
+        return new DecodedResponseFrame<>(apiVersion(), correlationId(), header, message);
```

Comment:
consider failing if `apiKey() != message.apiKey()`.  

essentially I am suggesting pushing code like this into DecodedRequestFrame
```
            if (message.apiKey() != decodedFrame.apiKeyId()) {
                throw new AssertionError(
                        "Filter '" + filterDescriptor() + "': Attempt to respond with ApiMessage of type " + ApiKeys.forId(message.apiKey()) + " but request is of type "
                                + decodedFrame.apiKey());
            }
```

Maybe add a unit test for DecodedRequestFrame